### PR TITLE
Fixes and factories

### PR DESCRIPTION
**Pull request by kirubel**
Started to work on valid_move but came across minor bugs here and there so fixed that.  Also setup factory tests for king, game, and user.  Also added Pry for easier debugging.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,6 +30,9 @@ gem 'coffee-rails', '~> 4.0.0'
 # Use jquery as the JavaScript library
 gem 'jquery-rails'
 
+group :development, :test do
+  gem 'pry-rails'
+end
```
Comment (washburnad) [inline Gemfile:35]: Great add!  If you guys don't know how to use pry look it up.  Great debugging tool.  Or perhaps pair with kirubel for a few minutes to let him teach you.

File: app/models/game.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class Game < ActiveRecord::Base
-	belongs_to :user
+	belongs_to :white_player, class_name: "User"
+	belongs_to :black_player, class_name: "User"
 	has_many :pieces
```
Comment (washburnad) [inline app/models/game.rb:4]: Good call.  However, I think we had decided on a has_many :users when we developed the spreadsheet and that might be a better way to go.

File: test/factories/pieces.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 FactoryGirl.define do
   factory :piece do
-    
+   association :game 
+  end
+  factory :king, parent: :piece do
   end
```
Comment (washburnad) [inline test/factories/pieces.rb:6]: Awesome to see the factories being built.

File: app/models/game.rb (hunk 2 above)
Comment (ghost) [inline app/models/game.rb:4]: quick question Aaron - Anna and I had discussed whether it should be belongs_to or has_many. In  our user.rb model we put  
class User < ActiveRecord::Base 
has_many :games

So doesn't this mean we have to have 
class Game<ActiveRecord::Base
belongs_to :user 
?

I remember reading that has_many and belongs_to should be paired, but I can't find it.

Comment (washburnad) [inline app/models/game.rb:4]: yes, they should complement each other.  Our database showed a game has_many :users and vice versa.  
the way this is written we can have @game.user - but who does this connect us to?  A game actually has 2 users (players) that we want to be able to connect it to.

**Pull request merged**